### PR TITLE
fix typehints

### PR DESCRIPTION
**Pull request by JaskRendix**
progressing #1211 mypy (1.1.1) --strict, now below 200
mostly because I was able to fix the typehint with pygame_menu, it was triggered by the list;
the other are imports;
forgot itertools in tools.py from yesterday PR;

Found 247 errors in 53 files (checked 291 source files)
Found 184 errors in 45 files (checked 291 source files)

tested, black and isort

### CODE REVIEW

**Review by kerizane — APPROVED** [review]

File: tuxemon/networking.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from tuxemon.middleware import Controller, Multiplayer
 from tuxemon.npc import NPC
 from tuxemon.session import local_session
-from tuxemon.states import world
+from tuxemon.states.world import WorldState
```
Comment (kerizane) [inline tuxemon/networking.py]: Do we need this change? I think this is causing the python3.10 tests error.

**Review by kerizane — CHANGES_REQUESTED** [review]
Fix tests

**Comment by JaskRendix** [comment]
No, I'm not going to fix the tests right now.
Found 186 errors in 45 files (checked 291 source files)

File: tuxemon/networking.py (hunk 1 above)
Comment (JaskRendix) [reply tuxemon/networking.py]: No, we don't need it at the moment.

**Review by kerizane — APPROVED** [review]

**Comment by kerizane** [comment]
I was asking for you to fix the tests in the python3.10 build, which you have done. All good.

**Pull request merged**